### PR TITLE
osquery 2.3.3

### PR DESCRIPTION
**Pull request by ilovezfs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

CC @theopolis

### CODE REVIEW

**Comment by theopolis** [comment]
Thanks! If you grab 2.3.3 you can keep support at mavericks. :)

**Comment by ilovezfs** [comment]
@theopolis cool. 2.3.3 supersedes 2.3.2 then despite the "pre-release" label?

**Comment by theopolis** [comment]
Also, I put some time into the cpp-Netlib and Boost 1.63 API incompatibilities issue and think I have a solution using small patches to each dependency. I'll keep you updated in the related issues.

**Comment by theopolis** [comment]
@ilovezfs, it does for macOS

**Comment by ilovezfs** [comment]
>it does for macOS

Great, I'll use that then.

>Also, I put some time into the cpp-Netlib and Boost 1.63 API incompatibilities issue and think I have a solution using small patches to each dependency. I'll keep you updated in the related issues.

Neat! Thank you.

File: Formula/osquery.rb [hunk 1]
```diff
@@ -80,6 +79,14 @@ class Osquery < Formula
     sha256 "61955afa09ef244fc84a72ef019de15515e76377aceeb2cbf1e93fa0df374cd2"
   end
 
+  # Remove for > 2.3.2
+  # Fix "error: 'shared_timed_mutex' is unavailable"
+  # Upstream commit from 15 Feb 2017 "darwin: Use boost shared_mutex for OS X"
+  patch do
+    url "https://github.com/facebook/osquery/commit/d1d21cd.patch"
```
Comment (theopolis) [inline Formula/osquery.rb]: Ah, for macOS this is essentially 2.3.3, so you can pull in the patch or use the pre-release tag, either way is fine.

File: Formula/osquery.rb [hunk 2]
```diff
@@ -16,8 +15,8 @@ class Osquery < Formula
 
   fails_with :gcc => "6"
 
-  # osquery only supports OS X 10.9 and above. Do not remove this.
-  depends_on :macos => :mavericks
+  # osquery only supports OS X 10.11 and above. Do not remove this.
+  depends_on :macos => :el_capitan
```
Comment (theopolis) [inline Formula/osquery.rb]: Is Yosemite an option?

Comment (ilovezfs) [inline Formula/osquery.rb]: Indeed.

**Comment by ilovezfs** [comment]
PR refreshed.

**Comment by ilovezfs** [comment]
Shipped.

**Pull request closed without merging**